.buildkite/release.yml

```
@@ -18,62 +18,111 @@ steps:
     agents:
       system: x86_64-linux
 
-  - label: 'Build package (linux)'
-    key: linux-package
-    depends_on: add-release-commits
-    command:
-      - ./scripts/buildkite/release/linux-package.sh
-    artifact_paths: [ "./result/linux/**" ]
-    agents:
-      system: ${linux}
+  - group: "Linux E2E"
+    key: "linux-e2e"
+    steps:
+      - block: 'Linux build and test'
+        depends_on: add-release-commits
+        key: linux-block
+        if: build.branch != "master"
 
-  - label: 'Run linux e2e tests'
-    key: linux-e2e
-    depends_on:
-      - add-release-commits
-      - linux-package
-    commands: |
-      ./scripts/buildkite/release/linux-e2e.sh
-    artifact_paths:
-      - "./result/linux/**"
-      - "./logs/**/*"
-    env:
-      NODE_STATE_DIR: "${STATE_DIR?}/node/preprod"
-    agents:
-      system: x86_64-linux
+      - label: 'Build package (linux)'
+        key: linux-package
+        depends_on:
+          - add-release-commits
+          - linux-block
+        command:
+          - ./scripts/buildkite/release/linux-package.sh
+        artifact_paths: [ "./result/linux/**" ]
+        agents:
+          system: x86_64-linux
 
-  - label: 'Build package (macOS, x86_64)'
-    key: macos-intel-package
-    depends_on:
-      - add-release-commits
-    command:
-      - ./scripts/buildkite/release/macos-intel-package.sh
-    artifact_paths:
-      - "./result/macos-intel/**"
-    agents:
-      system: ${macos}
+      - label: 'Run linux e2e tests'
+        depends_on:
+          - linux-package
+        commands: |
+          ./scripts/buildkite/release/linux-e2e.sh
+        artifact_paths:
+          - "./result/linux/**"
+          - "./logs/**/*"
+        env:
+          NODE_STATE_DIR: "${STATE_DIR?}/node/preprod"
+        agents:
+          system: x86_64-linux
 
-  - label: 'Build package (macOS, arm64)'
-    key: macos-silicon-package
-    depends_on:
-      - add-release-commits
-    command:
-      - ./scripts/buildkite/release/macos-silicon-package.sh
-    artifact_paths:
-      - "./result/macos-silicon/**"
-    agents:
-      system: ${macos}
+  - group: "Docker E2E"
+    key: "docker-e2e"
+    steps:
+      - block: 'Docker build and test'
+        depends_on: add-release-commits
+        key: docker-block
+        if: build.branch != "master"
 
-  - label: 'Run E2E tests (macOS, arm64)'
-    key: macos-silicon-e2e
-    depends_on:
-      - add-release-commits
-      - macos-silicon-package
-    command:
-      - nix develop path:./scripts/buildkite/release -c ./scripts/buildkite/release/macos-silicon-e2e.sh
-    artifact_paths:
-      - "./logs/**/*"
-    env:
-      NODE_STATE_DIR: "${STATE_DIR_MACOS?}/node/preprod"
-    agents:
-      system: ${macos}
+      - label: 'Build Docker Image'
+        key: build-docker
```

```suggestion
        key: docker-build
```

To make `key` = name of the shell script? 🤔